packages/language-server/src/utils/files/resolveTemplate.ts

```diff
@@ -0,0 +1,41 @@
+import { fileStat } from './fileStat';
+
+// HACK: Maybe we should move these to a configuration setting?
+const extensions = [`.twig`, `.html`];
+
+const indexTemplateFilenames = [`index`];
+
+/**
+ * Searches for a template file, and returns the first match if there is one.
+ */
+export const resolveTemplate = async (
+    pathToTwig: string,
+): Promise<string | null> => {
+    let stats = await fileStat(pathToTwig);
+
+    if (stats && stats.isFile()) {
+        return pathToTwig;
+    }
+
+    for (const extension of extensions) {
+        const testPath = `${pathToTwig}${extension}`;
+        stats = await fileStat(testPath);
+
+        if (stats?.isFile()) {
+            return testPath;
+        }
+    }
+
+    for (const filename of indexTemplateFilenames) {
+        for (const extension of extensions) {
+            const testPath = `${pathToTwig}/${filename}${extension}`;
+            stats = await fileStat(testPath);
+
+            if (stats?.isFile()) {
+                return testPath;
+            }
+        }
+    }
```

Comment:
nit: This loop over an array of one element kinda introduces complexity to the code. Could we just inline that `index` string and get rid of `for` over `indexTemplateFilenames`?

```ts
for (const extension of extensions) {
    const testPath = `${pathToTwig}/index${extension}`;
    stats = await fileStat(testPath);

    if (stats?.isFile()) {
        return testPath;
    }
}
```

Or extract a constant, something like `CRAFT_INDEX_TEMPLATE_NAME`, if you prefer this way, and reuse it here.

Reply:
I've seen some setups that uses both `index` and `default`, specially when either dealing with long running projects that have been worked by different agencies, or migrating an existing project to a new convention, which is why Craft has support for an array of index files names instead of just one, so projects can be transitioned over time.

Reply:
Let's keep this in mind and return back to it if someone needs it or when the `indexTemplateFilenames` gets implemented as a configuration setting